### PR TITLE
feat: add some missing functions for game controllers

### PR DESCRIPTION
**Pull request by BarrOff**
Hello

This pull request adds bindings for some missing gamecontroller functions.

### CODE REVIEW

**Comment by BarrOff** [comment]
@Araq is there something preventing a merge?
I need some of the added procs and would really appreciate if they were to be added.

File: src/sdl2/gamecontroller.nim [hunk 1]
```diff
@@ -241,6 +274,86 @@ proc getButton*(
...
+  importc: "SDL_GameControllerRumble".}
+  ##
+  ## Start a rumble effect on a game controller.
+  ##
+  ## Each call to this function cancels any previous rumble effect, and calling
+  ## it with 0 intensity stops any rumbling.
+  ##
+  ## `Returns` 0, or -1 if rumble isn't supported on this controller
+
+proc gameControllerRumbleTriggers*(gamecontroller: GameControllerPtr,
```
Comment (Araq) [inline src/sdl2/gamecontroller.nim:289]: Likewise `rumbleTriggers`.

File: src/sdl2/gamecontroller.nim [hunk 2]
```diff
@@ -241,6 +274,86 @@ proc getButton*(
...
+  ## calling it with 0 intensity stops any rumbling.
+  ##
+  ## Note that this is rumbling of the _triggers_ and not the game controller as
+  ## a whole. This is currently only supported on Xbox One controllers. If you
+  ## want the (more common) whole-controller rumble, use
+  ## `gameControllerRumble` instead.
+  ##
+  ## `Returns` 0, or -1 if trigger rumble isn't supported on this controller
+
+proc gameControllerHasLED*(gamecontroller: GameControllerPtr): Bool32 {.
```
Comment (Araq) [inline src/sdl2/gamecontroller.nim:304]: `hasLED`.

File: src/sdl2/gamecontroller.nim [hunk 3]
```diff
@@ -241,6 +274,86 @@ proc getButton*(
...
+  importc: "SDL_GameControllerHasLED".}
+  ## Query whether a game controller has an LED.
+  ##
+  ## \returns SDL_TRUE, or SDL_FALSE if this controller does not have a
+  ##          modifiable LED
+
+proc gameControllerHasRumble*(gamecontroller: GameControllerPtr): Bool32 {.
```
Comment (Araq) [inline src/sdl2/gamecontroller.nim:311]: etc.

File: src/sdl2/gamecontroller.nim [hunk 4]
```diff
@@ -241,6 +274,86 @@ proc getButton*(
   ##
   ## The button indices start at index `0`.
 
+proc gameControllerRumble*(gamecontroller: GameControllerPtr,
```
[the remark this answers is not in the document]
Comment (BarrOff) [reply src/sdl2/gamecontroller.nim]: Has been addressed by commit [81f18d9](https://github.com/nim-lang/sdl2/pull/159/commits/81f18d9ba321ddf8145cb7b9850ac2660bc9331e)

File: src/sdl2/gamecontroller.nim (hunk 1 above)
Comment (BarrOff) [reply src/sdl2/gamecontroller.nim]: Has been addressed by commit [81f18d9](https://github.com/nim-lang/sdl2/pull/159/commits/81f18d9ba321ddf8145cb7b9850ac2660bc9331e)

File: src/sdl2/gamecontroller.nim (hunk 2 above)
Comment (BarrOff) [reply src/sdl2/gamecontroller.nim]: Has been addressed by commit [81f18d9](https://github.com/nim-lang/sdl2/pull/159/commits/81f18d9ba321ddf8145cb7b9850ac2660bc9331e)

File: src/sdl2/gamecontroller.nim (hunk 3 above)
Comment (BarrOff) [reply src/sdl2/gamecontroller.nim]: Have been addressed by commit [81f18d9](https://github.com/nim-lang/sdl2/pull/159/commits/81f18d9ba321ddf8145cb7b9850ac2660bc9331e)

**Comment by BarrOff** [comment]
@Araq 

I renamed the procs I added and the ones who where there before to the requested naming scheme.

**Pull request merged**